src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -286,12 +312,22 @@ private static boolean startUp() {
   private static void runMainLoop() {
     Logger.LOG_DEBUG("Running main loop...");
 
-    // Grab latest data and send via MQTT
-    try {
-      CConnectorDataMgr.checkForHistoricalDataAndSend(mqttMgr);
-    } catch (Exception e) {
-      Logger.LOG_CRITICAL("Unable to grab latest data and send via MQTT.");
-      Logger.LOG_EXCEPTION(e);
+    // Grab latest data and send via MQTT (if measurements enabled)
+    boolean measurementsEnabled = true;
+    if (connectorMeasurementEnableControlTag != null) {
+      measurementsEnabled =
+          (connectorMeasurementEnableControlTag.getTagValueAsInt()
+              == CONNECTOR_MEASUREMENT_CONTROL_TAG_ENABLE_VALUE);
+    }
+    if (measurementsEnabled) {
```

Comment:
That is by design. If the measurement control tag can't be enabled, it displays a warning that you need to create it manually to control measurements and it will default to measurements being enabled.

Reply:
This makes sense, but the implementation could be simplified. All parts of code that are concerned with measurements enabled now need to get the tag control object, check if it's null, then check if it's true or false. They should just call a single method that returns true or false.  The function should hide all of the complexity of the tagControl.

Reply:
I see what you mean now. There are two places this was being accessed, and one of which requires the actual TagControl object to set the value of the tag, so that one can't be updated.

Reply:
I added a new method getConnectorMeasurementsEnabled() that is referenced instead of having the logic directly in the method.